crates/caclang/src/helpers.rs

```diff
@@ -60,8 +60,8 @@ pub(crate) fn parse_and_validate_ctx(
 ) -> anyhow::Result<(Node, u64)> {
     let mut expr_ctx: Vec<String> = Vec::new();
     let mut priority: u64 = 0;
-    for rule in ctx.split(AND_TOKEN).into_iter() {
-        let parts: Vec<&str> = rule.trim().split_whitespace().collect();
+    for rule in ctx.split(AND_TOKEN) {
+        let parts: Vec<&str> = rule.split_whitespace().collect();
```

Comment:
trim() is required here. Consider the statement
```
platform is android and client is 1mg
```
splitting on AND gives me
```
platform is android<space>
<space>client is 1mg
```
annoying to parse